docs/source/blocks/blocks-core/teaser.md

```diff
@@ -0,0 +1,152 @@
+
+## Style Wrapper Properties
+
+The Teaser has a styling property to align the image. The image can be aligned left (default), right and top.
+
+The styling wrapper properties can be extended using: {ref}`blocks-style-wrapper-label`
+
+## Variations
+
+By default, the Teaser have a default variation, but this can be extended using block variations
+
+The variations can be extended using: {ref}`extensions-block-variations`
+
+## Data Adapter
+
+The Teaser has a data adapter function that allows you to tap into the changes in the settings, and being able to issue changes in other settings fields.
+This is valuable in the Teaser block because it saves an internal cache of the target element.
+If you select the target, these values are updated. Also, when you update the target, these values remain (default behavior), but you can issue another behavior.
+
+The data adapter function is defined in the block's settings `dataAdapter`, so you can override it and add your own, if required.
+This is the default adapter (you should stick to his signature):
+
+```js
+import { isEmpty } from 'lodash';
+
+export const TeaserBlockDataAdapter = ({
+  block,
+  data,
+  id,
+  onChangeBlock,
+  value,
+}) => {
+  let dataSaved = {
+    ...data,
+    [id]: value,
+  };
+  if (id === 'href' && !isEmpty(value) && !data.title && !data.description) {
+    dataSaved = {
+      ...dataSaved,
+      title: value[0].Title,
+      description: value[0].Description,
+      head_title: value[0].head_title,
+    };
+  }
+  onChangeBlock(block, dataSaved);
+};
+
+```
+
+## Custom components in registry
+
+The Teaser block looks up for a couple of components in the registry, that you can provide in case that you want to modify the default behavior.
+
+### Image component
+
+The image component used in the Teaser can be overriden using your own custom image component.
+By default, it uses a simple `<img />` tag, with this signature:
+
+```jsx
+<Image
+  src={hasImageComponent ? href : defaultImageSrc}
+  alt=""
+  loading="lazy"
+/>
+```
+
+### Content type dependant Body component
+
+The Teaser component supports specifying a custom component depending on the target content type.
+
+You can provide your own component registering a component like in your add-on config:
+
+```js
+import NewsItem from './components/Blocks/Teaser/NewsItem';
+
+...
+
+  config.registerComponent({
+    name: 'Teaser',
+    component: NewsItem,
+    dependencies: 'News Item',
+  });
+```
+
+The `name` must be `Teaser` but then, as a dependency, you must provide the name of the content type.
+
+After this, if you select a content type `News Item` as target of the teaser, the `Body` component used will be `NewsItem` instead of the default one.
+
+## Custom Body component
+
+The Teaser body provides a default `Body` component that can be overriden.
+Although it can be done via component shadowing, it is recommended that you override it using the `variations` block settings field.
```

Comment:
```suggestion
Although it can be done via component shadowing, it is recommended that you override it using the `variations` block setting field.
```